### PR TITLE
update to neon gases and water quality munge kernels. started hbef retrieve kernels. 

### PR DESCRIPTION
**Pull request by spencerrhea**
@vlahm

### CODE REVIEW

File: src/global_helpers.R [hunk 1]
```diff
@@ -575,7 +575,8 @@ get_product_info <- function(network, domain=NULL, status_level, get_statuses){
 #. handle_errors
 prodcode_from_prodname_ms <- function(prodname_ms){
 
-    prodcode = strsplit(prodname_ms, '_')[[1]][2]
+    name_length = length(strsplit(prodname_ms, '_')[[1]])
+    prodcode = strsplit(prodname_ms, '_')[[1]][max(name_length)]
```
Comment (vlahm) [inline src/global_helpers.R:579]: nice catch. prodcodes that contain underscores would have been subdivided if we always grabbed the element at the second index. however, always grabbing the last won't do either. we want everything after the first. Say we encounter a chemistry prodcode "A_001". The prodname_ms would be "chemistry_A_001", and the stringsplit would separate the A and the 001, which should stay together. Also, this function isn't written to accept vector arguments, so name_length will always be a single integer. Using `max` is unnecessary.

File: src/lter/network_helpers.R [hunk 2]
```diff
@@ -39,11 +39,37 @@ get_avail_lter_product_sets <- function(prodname_ms, version, domain, data_track
     element_ids = reqdata[,2]
     dl_urls = paste0(dl_endpoint, domain_ref, '/', prodcode, '/', version,
         '/', element_ids)
-
+    
+    names <- str_match(reqdata[,3], '(.+?)_.*')[,2]
+    
+    #check to see if site information was included in lter informaiton
+    if(all(is.na(names))) {
```
Comment (vlahm) [inline src/lter/network_helpers.R:46]: here you account for the case where all names are NA, and below you account for the case where no names are NA. what about when some names are NA?

File: src/global_helpers.R [hunk 3]
```diff
@@ -576,7 +576,8 @@ get_product_info <- function(network, domain=NULL, status_level, get_statuses){
 prodcode_from_prodname_ms <- function(prodname_ms){
 
     name_length = length(strsplit(prodname_ms, '_')[[1]])
-    prodcode = strsplit(prodname_ms, '_')[[1]][max(name_length)]
+    prodcode = strsplit(prodname_ms, '_')[[1]][2:name_length]
+    prodcode <- paste(prodcode,collapse="_")
```
Comment (vlahm) [inline src/global_helpers.R:580]: nice : )

File: src/lter/network_helpers.R [hunk 4]
```diff
@@ -36,32 +36,37 @@ get_avail_lter_product_sets <- function(prodname_ms, version, domain, data_track
...
+    element_ids = reqdata[,2] 
     dl_urls = paste0(dl_endpoint, domain_ref, '/', prodcode, '/', version,
         '/', element_ids)
     
     names <- str_match(reqdata[,3], '(.+?)_.*')[,2]
     
-    #check to see if site information was included in lter informaiton
-    if(all(is.na(names))) {
+    #check to see if site information is included in lter informaiton
+    if(any(is.na(names))) {
```
Comment (vlahm) [inline src/lter/network_helpers.R:46]: muy righteoso

File: src/global_helpers.R [hunk 5]
```diff
@@ -749,3 +750,55 @@ update_product_statuses <- function(network, domain){
...
     return()
 }
+
+unit_converter <- function(val, input_unit, output_unit) {
+    
+    units <- tibble(prefix = c('n', "u", "m", "c", "d", "h", "k", "M"),
+        convert_factor = c(0.000000001, 0.000001, 0.001, 0.01, 0.1, 100, 
+            1000, 1000000))
+
+    old_fraction <- as.vector(str_split_fixed(input_unit, "/", n = Inf))
```
Comment (vlahm) [inline src/global_helpers.R:760]: did you add str_split_fixed to function_aliases.R? if not, gotta prepend `package_name::`

File: src/global_helpers.R [hunk 6]
```diff
@@ -749,3 +750,55 @@ update_product_statuses <- function(network, domain){
 
     return()
 }
+
+unit_converter <- function(val, input_unit, output_unit) {
```
Comment (vlahm) [inline src/global_helpers.R:754]: @spencerrhea next time we screenshare, let's see this puppy in action. @matthewross07, do you have a go-to unit conversion package?

File: src/global_helpers.R (hunk 5 above)
Comment (spencerrhea) [reply src/global_helpers.R:760]: I did not, but I will. Seems like it keeps coming up.

**Pull request merged**